app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4110,19 +4134,85 @@ public AppCompatActivity getParentActivity() {
         return (AppCompatActivity) ((ViewGroup) binding.getRoot().getParent()).getContext();
     }
 
-    private void useVideoSource(final boolean video) {
-        if (playQueue == null || isAudioOnly == !video || audioPlayerSelected()) {
+    private void useVideoSource(final boolean videoEnabled) {
+        if (playQueue == null || isAudioOnly == !videoEnabled || audioPlayerSelected()) {
             return;
         }
 
-        isAudioOnly = !video;
-        // When a user returns from background controls could be hidden
-        // but systemUI will be shown 100%. Hide it
+        isAudioOnly = !videoEnabled;
+        // When a user returns from background, controls could be hidden but SystemUI will be shown
+        // 100%. Hide it.
         if (!isAudioOnly && !isControlsVisible()) {
             hideSystemUIIfNeeded();
         }
+
+        // The current metadata may be null sometimes (for e.g. when using an unstable connection
+        // in livestreams) so we will be not able to execute the block below.
+        // Reload the play queue manager in this case, which is the behavior when the video stream
+        // played is not video only or when we don't know the index of the video renderer.
+        if (currentMetadata == null) {
+            reloadPlayQueueManager();
+            setRecovery();
+            return;
+        }
+
+        final int videoRenderIndex = getVideoRendererIndex();
+        final StreamInfo info = currentMetadata.getMetadata();
+
+        /* For video streams: we don't want to stream in background the video stream so if the
+        video stream played is not a video-only stream and if there is an audio stream available,
+        play this audio stream in background by reloading the play queue manager.
+        Otherwise the video renderer will be just disabled (because there is no
+        other stream for it to play the audio): if the video stream is video-only, only the audio
+        stream will be fetched and the video stream will be fetched again when the user return to a
+        video player.
+
+        For audio streams and audio live streams: nothing is done, it's not needed to reload the
+        player with the same audio stream.
+
+        For video live streams: the play queue manager is not reloaded if the stream source is a
+        live source (see VideoPlaybackResolver#resolve()) and if that's not the case, the
+        requirements for video streams is applied.
+
+        In the case where we don't know the index of the video renderer, the play queue manager
+        is also reloaded. */
```

Comment:
Idea: Binding this comment to actual code and not having a 10 line flowing text would be great and easier to read in the following code.